src/BatchWriter/Router.ts

```diff
@@ -26,22 +27,20 @@ export class Router {
   async start() {
     await this.messaging.consume(Exchange.ClaimIPFSHash, this.onClaimIPFSHash)
     await this.messaging.consume(Exchange.BatchWriterCreateNextBatchRequest, this.onBatchWriterCreateNextBatchRequest)
-    await this.messaging.consume(
```

Comment:
@kennylavender In my mind, the BatchWriter doesn't need to know anything about other systems. Its responsibility is to create batches and announce them, but ends there. It's up to other systems which want to do something with that Claim Batch to implement their own logic.

Reply:
Yeah. Your system made sense. I'm just thinking we could reduce overall complexity by giving BatchWriter limited responsibility.

The pro of your system is that in case of failure the directory that had been generated would be discarded, I think, and a new one created on the next cycle. That way we'd avoid queuing more and more timestamp requests if they fail.

My reasoning in that case was that we may want them to start piling up: if we just keep adding claims to the pending-batch queue, that's could be a sign of something going wrong or it could just be increased usage. But if timestamp requests start piling up, it's 100% guaranteed to be an issue and we can easily spot it and act on it.